download/download_test.go

```diff
@@ -109,3 +127,85 @@ func (suite *TestSuite) TestCreateFilePath() {
 	assert.NoError(suite.T(), err)
 }
+
+func (suite *TestSuite) TestGetURLsListFile() {
+
+	currentPath, err := os.Getwd()
+	assert.NoError(suite.T(), err)
+
+	// Folder URL does not exist
+	fileLocation := "https://some/base/A352744B-2CB4-4738-B6B5-BA55D25FB469/some/"
+
+	urlsFilePath, err := GetURLsListFile(currentPath, fileLocation)
+	assert.Equal(suite.T(), urlsFilePath, "")
+	// The error differs locally and in the repo, therefore checking that error starts
+	// with the specified phrase instead of the whole message
+	assert.True(suite.T(), strings.HasPrefix(err.Error(), "failed to download file, reason:"))
+
+	// File URL does not exist
+	fileLocation = "https://some/base/A352744B-2CB4-4738-B6B5-BA55D25FB469/some/urls_list.txt"
+
+	urlsFilePath, err = GetURLsListFile(currentPath, fileLocation)
+	assert.Equal(suite.T(), urlsFilePath, "")
+	// The error differs locally and in the repo, therefore checking that error starts
+	// with the specified phrase instead of the whole message
+	assert.True(suite.T(), strings.HasPrefix(err.Error(), "failed to download file, reason:"))
+
+	// File path
+	fileLocation = "some/path/to/urls_list.txt"
+	urlsFilePath, err = GetURLsListFile(currentPath, fileLocation)
+	assert.Equal(suite.T(), urlsFilePath, fileLocation)
+	assert.NoError(suite.T(), err)
+}
+
+func (suite *TestSuite) TestGetURLsListFilePass() {
+	urlsList := `http://url/to/file1.c4gh
+http://url/to/file2.c4gh
+http://url/to/file3.c4gh
+`
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, err := w.Write([]byte(urlsList))
+		assert.NoError(suite.T(), err)
+	}))
+	defer ts.Close()
+
+	file, err := os.Getwd()
+	if err != nil {
+		log.Printf("failed to get current directory, %v", err)
+	}
+
+	// Testing with url containing the file
+	fileLocation := ts.URL + "/A352744B-2CB4-4738-B6B5-BA55D25FB469/some/urls_list.txt"
+	urlsFilePath, err := GetURLsListFile(file, fileLocation)
+	assert.NoError(suite.T(), err)
+	// Check that the file exists
+	_, err = os.Stat(urlsFilePath)
+	assert.NoError(suite.T(), err)
+
+	// Check that the file contains the correct urls
+	expectedUrls, err := ioutil.ReadFile(urlsFilePath)
+	assert.NoError(suite.T(), err)
+	assert.Equal(suite.T(), expectedUrls, []byte(urlsList))
+
+	// Remove the file created from the downloadFile function
+	_ = os.Remove(urlsFilePath)
+
+	// Testing with the URL containing the file folder
+	fileLocation = ts.URL + "/A352744B-2CB4-4738-B6B5-BA55D25FB469/some/"
+	urlsFilePath, err = GetURLsListFile(file, fileLocation)
+	assert.NoError(suite.T(), err)
+
+	// Check that the file exists
+	_, err = os.Stat(urlsFilePath)
+	assert.NoError(suite.T(), err)
+
+	// Check that the file contains the correct urls
+	expectedUrls, err = ioutil.ReadFile(urlsFilePath)
+	assert.NoError(suite.T(), err)
+	assert.Equal(suite.T(), expectedUrls, []byte(urlsList))
+
+	// Remove the file created from the downloadFile function
+	_ = os.Remove(urlsFilePath)
+
+}
```

Comment:
This could have been one test, but eh, the most important is that we test all scenarios.